### PR TITLE
Added /ContentChannelItems/GetByCurrentPerson

### PR DESCRIPTION
**Pull request by asleepingpanda**
Adds a new API endpoint for content channel items that uses Rock's internal security process to filter out anything a current person should not be able to see.

### CODE REVIEW

**Comment by asleepingpanda** [comment]
Just wanted to check in on this and see if I could get just a rough timeline on when we think this could get merged/submitted to the Rock Shop?

**Comment by vinnyjth** [comment]
This current implementation checks to see if *all* items are approved for the current user, then it does all the filtering a user requested (by returning a queryable). Do you think you can find a way to have the OData filters apply first, before checking to see if a user can actually see the items? 

That might come with its own set of concerns, you might have to omit the `$limit` OData argument util after you have done your permissions filtering.

**Comment by asleepingpanda** [comment]
Hmmmm. That's a good catch and a great question.

So, basically if I were to query a collection with a limit of 5 items, but I only had permission to the 3 of those items, my query would result in my only seeing 3 items, right?

**Comment by vinnyjth** [comment]
As of right now, if you were to query a collection with 5 items, you would query all the items in your rock database, check every item for permission, then narrow down your results down to 5.

**Comment by vinnyjth** [comment]
Have you tested this locally?

**Comment by vinnyjth** [comment]
@asleepingpanda Local Rock I think seeds with maybe 20 content items, so I wouldn't be surprised if it was fast :) 

I think the trick here is going to be doing all the oData filtering _before_ you filter out by Authorization. I'm not sure how to do that honestly, because the oData filtering happens by default when you `return` a `Queryable()`. You won't be able to bundle the `isAuthorized` check inside the query, like the RocketChat person suggested, because of LINQ, it's not a valid expression and even if you can trick the typechecker you'll always get an error. (LINQ converts the method chain in a SQL call under the hood, and there's no valid SQL for `IsAuthorized`) You could try exploring a way to build a valid query that does the same thing that `a.IsAuthorized` does.

**Comment by asleepingpanda** [comment]
Ah ok that makes sense. I'll see what I can dig up on that `IsAuthorized` endpoint to see if I can figure out how to replicate that request.

So I pulled the code from this example inside of [Rock's own Rest API Endpoint](https://github.com/SparkDevNetwork/Rock/blob/develop/Rock.Rest/Controllers/ContentChannelItemsController.Partial.cs#L48) for querying Content Channel Item Children. Would this same inefficiency you're describing also be something that is active here that we could contribute back to Rock? Or is this not an issue in this given context due to them already having a layer of specificity with `id` being passed in?

**Comment by vinnyjth** [comment]
Probably less of an issues because you’re always restricted to an item’s children (rather than always querying all items).

**Comment by asleepingpanda** [comment]
Ok so it looks like we're in a bit of rut here... When asking some of the Rock devs about this, they were explaining that security is handled in memory and _cannot_ be handled in SQL. The exact response is here:

> Correct. Not possible. The reason is security inheritance. SQL can't calculate the inheritance tree the authorization uses because it can be different for each entity type.
> 
> The security data itself is cached in memory, so checking security on the entities once they are loaded is rediculously fast. But you still have to load them into memory to perform the check. This is why most API methods implement security based on the access to the API endpoint rather than access to the actual entities themselves.

And when clarifying with them on all possible solutions, they assured me that the only way that they know how would be to do as we've discussed here which would be to query all `ContentChannelItems`, run the security filter, then run the `Queryable` on that filtered collection.

Since the context of the Apollos Project only requires access to a handful of Content Channels, perhaps having a required parameter of `ContentChannelIds` could help limit the scope of items that need to be queried (similar to the Children example from above)

As the conversation continues on RocketChat, the devs also suggested looking into something [like this](https://docs.microsoft.com/en-us/aspnet/web-api/overview/odata-support-in-aspnet-web-api/supporting-odata-query-options#invoking-query-options-directly) in order to get even deeper into the OData stack.

So for speed of development, I feel like requiring a `ContentChannelId` parameter would probably be the easiest solution at this point. What are your thoughts @vinnyjth ?

**Comment by conrad-vanl** [comment]
> So for speed of development, I feel like requiring a ContentChannelId parameter would probably be the easiest solution at this point. What are your thoughts @vinnyjth ?

@asleepingpanda I can get on board with this option, that sounds good to me.

Also, please not that if this isn't ready by EoD, we will be closing it. You can re-open it when it's ready though. This isn't you, it's us: I'm trying to make sure our team is responsive to PRs by giving a hard deadline for PRs to get in and we flush everything out. The important things will surface back up, like this one 😉

But also, you can rest assured that if you can make the suggested updates today, it will get merged in!

**Comment by asleepingpanda** [comment]
In that case, working on it literally right now. Lol

**Comment by asleepingpanda** [comment]
@vinnyjth @conrad-vanl Updated this per the conversation and also per some C# best practices suggested by the Rock team. Go easy on me! I haven't written in C# in like 2 or 3 years, now. I kept on writing `const` 🤦  Hahah.

**Comment by conrad-vanl** [comment]
@asleepingpanda I think this looks good to me, I'll let @vinnyjth do the final review today when he's back.

**Comment by asleepingpanda** [comment]
It's 5:38pm EST.... DON'T CLOSE MY PR!! I GOT IT IN TIME! lol

File: apollosproject.ApollosPlugin/Rest/ApollosController.cs [hunk 1]
```diff
@@ -345,5 +345,49 @@ public IQueryable<ContentChannelItem> ContentChannelItemsByAttributeValue(string
...
+        {
+            // Split the given string of Content Channel Ids into a list and query for 
+            // Content Channel Items inside of those Content Channels
+            List<int> contentChannelIdList = contentChannelIds.Split(',').Select(int.Parse).ToList();
+
+            // If no Content Channel Ids were given, just return an empty result so make
+            // life easier for Rock
+            if (contentChannelIdList.Count < 1)
+            {
+                return Enumerable.Empty<ContentChannelItem>().AsQueryable();
```
Comment (vinnyjth) [inline apollosproject.ApollosPlugin/Rest/ApollosController.cs:369]: Nice!

**Pull request merged**